### ISSUE TITLE
Migration guide is a mess. Can we create an actual migration guide?

### ISSUE THREAD

**Issue by jsilveira** [sveltejs/kit#9532]
### Describe the problem

Context: My first migration of an old Svelte Kit app took several hours for a very small project. The steps were not clear and I found the guide confusing and incomplete, coming across several errors that took a while to google and troubleshoot. Weeks later on my second migration of a larger app I thought things would be smooth given my previous experience. But again, I came across more unexpected issues and outdated/poor documentation. After #9523 and reading several experiences like mine (while googling how to solve my migration issues) I realized this had not been properly brought up or discussed, so I'm creating this issue.

The problems: Current migration guide is like a treasure hunt. A discussion, with a link to a comment, with a link to a PR, with a link to an issue. None of it is writen in a migration doc fashion because the purpose of those PRs wasn't to be documentation of a migration process.  It's confusing, painful and a huge waste of time for all the community members that are not able to read every PR of Svelte and still don't have all the (recently changed) know-how of how things work now in their head.  It's painful enough for early adopters of Kit to keep up with all the 1.x changes, but can't we at least have a thoroughly written migration guide that at least mentions in an ordered fashion all the breaking changes in one place? 

Some examples of things that I found confusing/missing/broken in the current guide:
- Hard to follow in what order to update to which version. To .405 before or after npx svelte-migrate? 
- Problems with conflicting Vite versions when updating
- Vite config changes buried in PR
- Renames not in the guide (such as app/env -> app/environment, %svelte.(body,head,assets)% to %sveltekit.(body,head,assets)%)
- Changing all my POST/UPDATE/DELETE endpoints according to the message indicated my the @migration annotation to later find out it was outdated and all the code changes were a waste of time (see #9523).  

Also, a list of github issues marked with "breaking change" is not useful, what am I supposed to read 111 closed issues? And a changelog with 4884 lines is not a migration guide either. It's hard to reconstruct from there all the things that changed. 

I know most of these issues are trivial to solve once you understand what is happening, but a migration process shouldn't have to be a process of Googling ten different errors and dissecting the github history of a project. It should be a boring (and maybe long) list of small steps that if you follow them in order it takes you to a working app.

### Describe the proposed solution

I know It's not fun investing time in writing doc meant only for a (small?) and fixed group of early adopters who used Svelte Kit in the past and that will only be useful for a couple of years at most, but I think it is fair to say that those early users adoption made Svelte Kit 1.0 something worth building and so they deserve a proper migration guide.

### Importance

would make my life easier

**Comment by CaptainCodeman** [comment]
Personally, I think this is unreasonable and I'd much rather people be spending their effort on improving the current released version than polishing the docs for unreleased versions that every day becomes less relevant and applicable to fewer people.

Part of the deal with using pre-release software is that you commit to tracking the project and the changes as it evolves. Creating an app with an old version and then coming back to it to update a long time later is difficult as it's unreasonable to expect an upgrade guide to the released version from every single pre-release change along the way.

The information is there, I've used the migrate tool on at least a dozen projects, if not more. The migration tool was for the most impactful change to the routing, and it does most of the repetitive work for you. The guide was to use that to get past that biggest version change, not to tell you how to get to v1 from any previous version.

If you get stuck, I'd suggest asking for help in Discord.

**Comment by dummdidumm** [comment]
I think the main problem is that many people try to bump their version directly to 1.0 and go through all the migrations at once. The migration docs which are there are clearly not designed with that in mind - nor should they, since it's impossible to write migration steps for all possible starting/ending points. Instead we should emphasize in the main migration doc that you should do the migration step by step, or in other words upgrade to version X with some breaking changes, then version Y with some more breaking changes etc, so that each of the migration docs (which are written out on the PRs mostly) can be easily followed.

So what we'll do is adjust the main migration doc to emphasize a) the importance of step-by-step more and b) clearly state which versions to migrate to and which PRs to look at which contain the important migrations.

**Comment by jsilveira** [comment]
I understand, but if you upgrade step by step wouldn't you have to rewrite some parts of your code more than once with breaking changes later overruled by new breaking changes? Thinking of `page.server.js` for example. 

Would it make sense to make a more complete list of the most common unavoidable breaking changes from 1.0.0-next-406 to the current version? I feel like most breaking changes came after that version, but maybe I'm wrong on this one.
